python/cugraph/cugraph/structure/property_graph.py

```diff
@@ -1748,6 +1928,180 @@ def renumber_edges_by_type(self):
+        missing : {"ignore", "error"}
+            If "ignore", empty or null rows without vector data will be skipped
+            when creating the array, so output array shape will be
+            [# of non-empty rows] by [size of vector property].
+            When "error", RuntimeError will be raised if there are any empty rows.
+            Ignored if fillvalue is given.
+
+        Returns
+        -------
+        cupy.ndarray or numpy.ndarray
+        """
+        if col_name not in self.__edge_vector_property_lengths:
+            raise ValueError(f"{col_name!r} is not a known edge vector property")
+        length = self.__edge_vector_property_lengths[col_name]
+        return self._get_vector_property(df, col_name, length, fillvalue, missing)
+
+    def _check_vector_properties(
+        self, df, vector_properties, vector_property_lengths, invalid_keys
+    ):
+        """Check if vector_properties is valid and update vector_property_lengths"""
+        df_cols = set(df.columns)
+        for key, columns in vector_properties.items():
+            if key in invalid_keys:
+                raise ValueError(
+                    "Cannot assign new vector property to existing "
+                    f"non-vector property: {key}"
+                )
+            if isinstance(columns, str):
+                # If df[columns] is a ListDtype column, should we allow it?
+                raise TypeError(
+                    f"vector property columns for {key!r} should be a list; "
+                    f"got a str ({columns!r})"
+                )
+            if not df_cols.issuperset(columns):
+                missing = ", ".join(set(columns) - df_cols)
+                raise ValueError(
+                    f"Dataframe does not have columns for vector property {key!r}:"
+                    f"{missing}"
+                )
+            if not columns:
+                raise ValueError("Empty vector property columns for {key!r}!")
+            if vector_property_lengths.get(key, len(columns)) != len(columns):
+                prev_length = vector_property_lengths[key]
+                new_length = len(columns)
+                raise ValueError(
+                    f"Wrong size for vector property {key}; got {new_length}, but "
+                    f"this vector property already exists with size {prev_length}"
+                )
+        for key, columns in vector_properties.items():
+            vector_property_lengths[key] = len(columns)
+
+    def _create_vector_properties(self, df, vector_properties):
+        if self.__series_type is cudf.Series:
+            ascontiguousarray = cupy.ascontiguousarray
+        else:
+            ascontiguousarray = np.ascontiguousarray
+        # Make each vector contigous and 1-d
+        vectors = {
+            key: [
+                np.squeeze(vec, 0)
+                for vec in np.split(ascontiguousarray(df[columns].values), len(df))
+            ]
```

Comment:
I suggest using `NEP 35` and using numpy with like here ? https://numpy.org/neps/nep-0035-array-creation-dispatch-with-array-function.html

Reply:
Yeah, that's nicer--thanks!

Reply:
Done. Had change to a for loop instead of a dict comprehension, but I think it's a little clearer.